fvtest/compilertriltest/LinkageTest.cpp

```diff
@@ -151,3 +150,291 @@ TYPED_TEST(LinkageTest, SystemLinkageParameterPassingFourArg) {
+                                                  TypeToString<TypeParam>::type,   // Args
+                                                  TypeToString<TypeParam>::type,   // Args
+                                                  TypeToString<TypeParam>::prefix, // return
+                                                  TypeToString<TypeParam>::prefix, // call
+                                                  reinterpret_cast<uintmax_t>(static_cast<TypeParam (*)(TypeParam,TypeParam,TypeParam,TypeParam,TypeParam)>(stackUser<TypeParam>)),// address
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::prefix, // load
+                                                  TypeToString<TypeParam>::prefix, // load
+                                                  TypeToString<TypeParam>::prefix, // load
+                                                  TypeToString<TypeParam>::prefix, // load
+                                                  TypeToString<TypeParam>::prefix  // load
+                                                  );
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees) << "Trees failed to parse\n" << inputTrees;
+
+    // Execution of this test is disabled on non-X86 platforms, as we
+    // do not have trampoline support, and so this call may be out of
+    // range for some architectures.
+#ifdef TR_TARGET_X86
+    Tril::DefaultCompiler compiler{trees};
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+
+    auto entry_point = compiler.getEntryPoint<TypeParam (*)(TypeParam,TypeParam,TypeParam,TypeParam,TypeParam)>();
+
+    EXPECT_EQ(static_cast<TypeParam>(2053),     stackUser<TypeParam>(1,1,1,1,static_cast<TypeParam>(1024)))     << "Input Trees: " << inputTrees;
+    EXPECT_EQ(static_cast<TypeParam>(3),        stackUser<TypeParam>(1,1,1,1,static_cast<TypeParam>(-1)))       << "Input Trees: " << inputTrees;
+    EXPECT_EQ(static_cast<TypeParam>(0x1e1e23), stackUser<TypeParam>(1,1,1,1,static_cast<TypeParam>(0xf0f0f)))  << "Input Trees: " << inputTrees;
+
+    EXPECT_EQ(static_cast<TypeParam>(2053),     entry_point(1,1,1,1,static_cast<TypeParam>(1024)))     << "Input Trees: " << inputTrees;
+    EXPECT_EQ(static_cast<TypeParam>(3),        entry_point(1,1,1,1,static_cast<TypeParam>(-1)))       << "Input Trees: " << inputTrees;
+    EXPECT_EQ(static_cast<TypeParam>(0x1e1e23), entry_point(1,1,1,1,static_cast<TypeParam>(0xf0f0f)))  << "Input Trees: " << inputTrees;
```

Comment:
I'm not sure I understand this. Isn't the point of `stackUser()` to do something like

```c++
EXPECT_EQ(stackUser<TypeParam>(1,1,1,1,static_cast<TypeParam>(1024)),
          entry_point(1,1,1,1,static_cast<TypeParam>(1024)))
      << "Input Trees: " << inputTrees;
```

Reply:
I've add a comment, the goal is just to test the fact the compiled function itself works fine.

Reply:
Hm... Ok, I see.